crypto/mem_sec.c

```diff
@@ -493,6 +527,9 @@ static int sh_init(size_t size, size_t minsize)
             ret = 2;
         }
     }
+#elif defined(_WIN32)
+    if (VirtualLock(sh.arena, sh.arena_size) == FALSE)
```

Comment:
It seems there could be problems here. According to the VirtuaLock documentation:

"The maximum number of pages that a process can lock is equal to the number of pages in its minimum working set minus a small overhead."

And according to the SetProcessWorkingSetSize documentation for `dwMinimumWorkingSetSize`:

"The default size is 50 pages (for example, this is 204,800 bytes on systems with a 4K page size)."

So, it seems a typical value for the maximum amount of secure memory in a process might be 200k. I have no idea how much secure memory a typical process uses, but I wonder if it ever approaches 200k? Perhaps not, but I wonder if we should document this limitation somewhere (probably on the `CRYPTO_secure_malloc_init` man page) and how to increase this?

Reply:
@richsalz, any idea about the sizing here?

Linux/Unix isn't going to permit unlimited size either.  `ulimit -l` can restrict it further.

Reply:
You'll have to do some estimating, based on the number of private keys and the number of simultaneous sessions you expect to have.  There is no good answer.  Perhaps adding a couple of sentences like that to the manpage would be useful.  The size we use is far above what would be typical for a common origin 😃

Reply:
Yeah, as someone who consumes OpenSSL's secure heap from a managed language, on Linux the default is 64KB, so the situation is actually much worse on other platforms.

Reply:
Given that the default limit for Windows is larger than the limit for Linux, I'm inclined to consider this a non-issue, but I'd love more feedback.  We can certainly adjust the minimum and maximum working set size if we think that makes sense, but we'd likely want to do the same setrlimit call on other platforms too.  Personally given the comparison to Linux defaults, I'd make introducing limit adjustments to be another effort.

Reply:
Ok. Let's leave it at that for this PR. Wouldn't it be nice if the arena size could dynamically expand and contract as required? If we had that feature then this could probably be configured by default through the auto initialisation rather than requiring explicit init and then all users could benefit from it.